source/agora/common/BitMask.d

```diff
@@ -0,0 +1,195 @@
+/*******************************************************************************
+
+    This is an implementation of a bitmask that once initialized will have a
+    fixed number of bits for holding `true` / `false` (`1` / `0`) values.
+    It allocates the required number of `ubytes` in the constructor and does not
+    allow reading or writing to bits beyond the fixed count which is set during
+    construction.
+    This type is created for use as the validators signing bitmask and any
+    changes should ensure that it does not compromise that use.
+
+*******************************************************************************/
+
+module agora.common.BitMask;
+
+import agora.serialization.Serializer;
+
+public struct BitMask
+{
+    import std.algorithm;
+    import std.conv;
+    import std.range;
+
+    @safe pure:
+
+    /// String serialization
+    public string toString () const
+    {
+        return iota(this.count).map!(i => this[i] ? "1" : "0").join("");
```

Comment:
As discussed IRL: `"4/6 (2, 4)"`, where the number between parenthesis is 0-indexed.
Also needs to use a sink.

Reply:
Will update. Just getting the tests to pass first.

Reply:
This will be done for PrettyPrint only as for vibe.d we will transfer as string of bit values (e.g. `01001111`)